### PR TITLE
[CognitiveServices] Enable Bing Search/CustomSearch products to specify Endpoint 

### PR DESCRIPTION
**Pull request by yangyuan**
## Description
Enable Bing Search/CustomSearch products to specify Endpoint 
https://github.com/Azure/azure-rest-api-specs/pull/4401

There is no API changes.

| API               | Breaking Change   | Version        |
|---|---|---|
| SpellCheck        | One constructor deleted | 3.0.0 -> 4.0.0 |
| NewsSearch        | client name change | 1.2.0 -> 2.0.0 |
| VisualSearch      | client name change | 1.2.0 -> 2.0.0 |
| EntitySearch      | client name change | 1.2.0 -> 2.0.0 |
| CustomSearch      | client name change | 1.3.0 -> 2.0.0 |
| VideoSearch       | client name change | 1.2.0 -> 2.0.0 |
| AutoSuggest       | client name change | 1.0.0 -> 2.0.0 |
| CustomImageSearch | client name change | 1.2.0 -> 2.0.0 |
| WebSearch         | client name change | 1.2.0 -> 2.0.0 |
| ImageSearch       | client name change | 1.2.0 -> 2.0.0 |

For you to easier to review, I break down the changes in to 2 parts.
* First one is just updating auto-generated code.
* Second one is version bump/class name update in tests. 

---

This checklist is used to make sure that common guidelines for a pull request are followed.
- [x] Please add REST spec PR link to the SDK PR
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [ ] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [x] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [x] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [x] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

File: src/SDKs/CognitiveServices/dataPlane/Search/BingCustomImageSearch/BingCustomImageSearch/Generated/CustomImageSearch/CustomImageSearchClient.cs [hunk 1]
```diff
@@ -210,75 +172,6 @@ public CustomImageSearchClient(ServiceClientCredentials credentials, HttpClientH
...
-        /// <param name='rootHandler'>
-        /// Optional. The http client handler used to handle http transport.
-        /// </param>
-        /// <param name='handlers'>
-        /// Optional. The delegating handlers to add to the http client pipeline.
-        /// </param>
-        /// <exception cref="System.ArgumentNullException">
-        /// Thrown when a required parameter is null
-        /// </exception>
-        public CustomImageSearchClient(System.Uri baseUri, ServiceClientCredentials credentials, HttpClientHandler rootHandler, params DelegatingHandler[] handlers) : this(rootHandler, handlers)
```
Comment (dsgouda) [inline src/SDKs/CognitiveServices/dataPlane/Search/BingCustomImageSearch/BingCustomImageSearch/Generated/CustomImageSearch/CustomImageSearchClient.cs]: This is a breaking change, needs major version bump

File: src/SDKs/CognitiveServices/dataPlane/Search/BingCustomSearch/BingCustomSearch/Generated/CustomSearch/CustomSearchClient.cs [hunk 2]
```diff
@@ -47,116 +54,56 @@ public partial class CustomSearchAPI : ServiceClient<CustomSearchAPI>, ICustomSe
...
-        /// <param name='credentials'>
-        /// Required. Subscription credentials which uniquely identify client subscription.
-        /// </param>
-        /// <param name='handlers'>
-        /// Optional. The delegating handlers to add to the http client pipeline.
-        /// </param>
-        /// <exception cref="System.ArgumentNullException">
-        /// Thrown when a required parameter is null
-        /// </exception>
-        public CustomSearchAPI(ServiceClientCredentials credentials, params DelegatingHandler[] handlers) : this(handlers)
```
Comment (dsgouda) [inline src/SDKs/CognitiveServices/dataPlane/Search/BingCustomSearch/BingCustomSearch/Generated/CustomSearch/CustomSearchClient.cs]: Breaking change, please bump the nuget major version

**Review by dsgouda — CHANGES_REQUESTED** [review]
Please update the nuget versions
Any chance this could be broken into smaller PRs?

File: src/SDKs/CognitiveServices/dataPlane/Search/BingCustomSearch/BingCustomSearch/Generated/CustomSearch/CustomSearchClient.cs (hunk 2 above)
Comment (yangyuan) [reply src/SDKs/CognitiveServices/dataPlane/Search/BingCustomSearch/BingCustomSearch/Generated/CustomSearch/CustomSearchClient.cs]: Done

File: src/SDKs/CognitiveServices/dataPlane/Search/BingCustomImageSearch/BingCustomImageSearch/Generated/CustomImageSearch/CustomImageSearchClient.cs (hunk 1 above)
Comment (yangyuan) [reply src/SDKs/CognitiveServices/dataPlane/Search/BingCustomImageSearch/BingCustomImageSearch/Generated/CustomImageSearch/CustomImageSearchClient.cs]: Done

**Comment by yangyuan** [comment]
Sorry didn't catch that two breaking changes.

This is a Cognitive Service scope PR with no API changes, breaking down into 10 individual PRs would be a lot of efforts but very limited benefits. So if it is okay I prefer we can just use this PR to cover all the participating APIs..

File: src/SDKs/CognitiveServices/dataPlane/Language/SpellCheck/BingSpellCheck/Generated/SpellCheckClient.cs [hunk 3]
```diff
@@ -144,54 +119,20 @@ public SpellCheckClient(ServiceClientCredentials credentials, params DelegatingH
...
-        /// <param name='rootHandler'>
-        /// Optional. The http client handler used to handle http transport.
-        /// </param>
-        /// <param name='handlers'>
-        /// Optional. The delegating handlers to add to the http client pipeline.
-        /// </param>
-        /// <exception cref="System.ArgumentNullException">
-        /// Thrown when a required parameter is null
-        /// </exception>
-        public SpellCheckClient(ServiceClientCredentials credentials, HttpClientHandler rootHandler, params DelegatingHandler[] handlers) : this(rootHandler, handlers)
```
Comment (dsgouda) [inline src/SDKs/CognitiveServices/dataPlane/Language/SpellCheck/BingSpellCheck/Generated/SpellCheckClient.cs]: This is a breaking change, needs major version bump

**Review by dsgouda — CHANGES_REQUESTED** [review]
Looks good apart from one required change

**Review by dsgouda — APPROVED** [review]
LGTM

**Comment by dsgouda** [comment]
Will merge on CIs passing

**Comment by yangyuan** [comment]
cool, thanks... looks all pass now.

**Pull request merged**